components/mediation-ui/org.wso2.carbon.rest.api.ui/src/main/resources/web/api/editapi-ajaxprocessor.jsp

```diff
@@ -31,47 +31,59 @@
 
 
 <%
-	ResourceBundle bundle = ResourceBundle.getBundle(
-			"org.wso2.carbon.rest.api.ui.i18n.Resources",
-			request.getLocale());
-	String url = CarbonUIUtil.getServerURL(this.getServletConfig()
-		.getServletContext(), session);
-	ConfigurationContext configContext = (ConfigurationContext) config
-		.getServletContext().getAttribute(
-				CarbonConstants.CONFIGURATION_CONTEXT);
-	String cookie = (String) session
-		.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
-	RestApiAdminClient client = new RestApiAdminClient(
-		configContext, url, cookie, bundle.getLocale());
+    ResourceBundle bundle = ResourceBundle.getBundle(
+            "org.wso2.carbon.rest.api.ui.i18n.Resources",
+            request.getLocale());
+    String url = CarbonUIUtil.getServerURL(this.getServletConfig()
+            .getServletContext(), session);
+    ConfigurationContext configContext = (ConfigurationContext) config
+            .getServletContext().getAttribute(
+                    CarbonConstants.CONFIGURATION_CONTEXT);
+    String cookie = (String) session
+            .getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
+    RestApiAdminClient client = new RestApiAdminClient(
+            configContext, url, cookie, bundle.getLocale());
 
-	String apiName = request.getParameter("apiName");
-	String apiContext = request.getParameter("apiContext");
-	String filename = request.getParameter("filename");
-	String hostname = request.getParameter("hostname");
-	String port = request.getParameter("port");
-	
-	if(port == null || "".equals(port)){
-		port = "-1";
-	}
-	
-	List<ResourceData> resourceList = 
-			(ArrayList<ResourceData>)session.getAttribute("apiResources");
-	APIData apiData = new APIData();
-	apiData.setName(apiName);
-	apiData.setContext(apiContext);
-	apiData.setHost(hostname);
-	apiData.setPort(Integer.parseInt(port));
-	apiData.setFileName(filename);
-	ResourceData resources[] = new ResourceData[resourceList.size()];
-	
-	if(resourceList != null) {
+    String apiName = request.getParameter("apiName");
+    String apiContext = request.getParameter("apiContext");
+    String filename = request.getParameter("filename");
+    String hostname = request.getParameter("hostname");
+    String port = request.getParameter("port");
+    String version = request.getParameter("version");
+    String versionType = request.getParameter("versionType");
+
+    if ("none".equals(versionType)) {
```

Comment:
Shouldn't we make "none" a constant? Even if its used once, making it a constant would improve maintainability.